### DISCUSSION TITLE
Create button card to change mop intensity

### DISCUSSION THREAD

**Discussion by smartmatic** [humbertogontijo/homeassistant-roborock#281] [Q&A]
Hi,

i've created a button in order to change the mop intensity. That's the code

show_name: true
show_icon: true
type: button
tap_action:
  action: call-service
  service: roborock.vacuum_set_mop_intensity
  target:
    device_id: f17b2d390e36b4832e6c3499df1b6f3e
  data: {}
name: Wischstärke ändern
icon: mdi:robot-vacuum
show_state: true

Button Card 

![button card](https://user-images.githubusercontent.com/74932663/236154675-2ca1d6f1-6650-46d9-a06e-a2514b84c308.png)

But when i try to call the service i got this error

![error_mop_intensity](https://user-images.githubusercontent.com/74932663/236154357-af27a03d-07db-4f5b-8941-41eb1b0c3aa3.png)

Ì have currently no idea what is missing to get it working?

**Comment by Lash-L** [comment]
You can just upgrade to the main branch/ wait for us to officially release it. It has the mop intensity as a select entity

**Reply by smartmatic** [reply]
With upgrade you mean change the files in my custom_integration folder to the latest version?
When will be the official release? I thought the integrations is since HA release 2023.5 officially rleased?

**Reply by Lash-L** [reply]
Yes - If you open HACS, you can hit redownload on roborock, then select 'main'. That will bring you up to main.

We are planning to release main as a 2.0 version soon, but we keep adding more features so it has been slowing us down.
